### PR TITLE
EOS-19028: Distribute config.sls phase-wise

### PR DESCRIPTION
**Pull request by tanujashinde0405**
Signed-off-by: Tanuja Shinde <tanuja.shinde@seagate.com>

### CODE REVIEW

File: srv/components/csm/config/config.sls [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+include:
+    - components.csm.config.post_install
```
Comment (divya-vijay) [inline srv/components/csm/config/config.sls:19]: @tanujashinde0405 I think we're calling all phases in `init_mod` . Please follow either of this - 
1. include the flow of all phases in init, or 
2. include each phase inside the next phase's file. (based on `require` steps)

Overall, better to follow step-1, to maintain uniformity and file discipline.

File: srv/components/csm/config/init_mod.sls [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+include:
+    - components.csm.config.config
+    - components.csm.config.post_install
```
Comment (divya-vijay) [inline srv/components/csm/config/init_mod.sls:20]: Overall, `post_install` is called first and `config` next. Please check the flow and update.

File: srv/components/csm/init.sls [hunk 3]
```diff
@@ -19,7 +19,9 @@
 include:
   - components.csm.prepare
   - components.csm.install
-  - components.csm.config
+  - components.csm.config.init_mod
+ # - components.csm.config.post_install
+ # - components.csm.config.config
```
Comment (divya-vijay) [inline srv/components/csm/init.sls:24]: If they're included inside `init_mod`, we can delete these lines from common `init`, instead of commenting.

**Review by divya-vijay — COMMENTED** [review]
@tanujashinde0405 Good work on your first PR! Left some minor comments to address.

File: srv/components/ha/cortx-ha/init.sls [hunk 4]
```diff
@@ -19,7 +19,7 @@
 include:
   - components.ha.cortx-ha.prepare
   - components.ha.cortx-ha.install
-  - components.ha.cortx-ha.config
+  - components.ha.cortx-ha.config.init_mod
```
Comment (ypise) [inline srv/components/ha/cortx-ha/init.sls:22]: Should be `components.ha.cortx-ha.config`, no?
Also, where is `srv/components/ha/cortx-ha/config/init.sls` that includes the 3 config sub-modules?
We need it until we make the switch to CES workflow

File: srv/components/hare/init.sls [hunk 5]
```diff
@@ -18,7 +18,7 @@
 include:
   - components.hare.prepare
   - components.hare.install
-  - components.hare.config
+  - components.hare.config.init_mod
```
Comment (ypise) [inline srv/components/hare/init.sls:21]: Should be components.hare.config, no?
Also, where is srv/components/hare/config/init.sls that includes the 3 config sub-modules?
We need it until we make the switch to CES workflow

File: srv/components/motr/init.sls [hunk 6]
```diff
@@ -19,7 +19,7 @@
 include:
   - components.motr.prepare
   - components.motr.install
-  - components.motr.config
+  - components.motr.config.init_mod
```
Comment (ypise) [inline srv/components/motr/init.sls:22]: Should be components.motr.config, no?
Also, where is srv/components/motr/config/init.sls that includes the 3 config sub-modules?
We need it until we make the switch to CES workflow

File: srv/components/s3server/init.sls [hunk 7]
```diff
@@ -19,7 +19,7 @@
 include:
   - components.s3server.prepare
   - components.s3server.install
-  - components.s3server.config
+  - components.s3server.config.init_mod
```
Comment (ypise) [inline srv/components/s3server/init.sls:22]: Should be components.s3server.config, no?
Also, where is srv/components/s3server/config/init.sls that includes the 3 config sub-modules?
We need it until we make the switch to CES workflow

File: srv/components/sspl/init.sls [hunk 8]
```diff
@@ -19,7 +19,7 @@
 include:
   - components.sspl.prepare
   - components.sspl.install
-  - components.sspl.config
+  - components.sspl.config.init_mod
```
Comment (ypise) [inline srv/components/sspl/init.sls:22]: Should be components.sspl.config, no?
Also, where is srv/components/sspl/config/init.sls that includes the 3 config sub-modules?
We need it until we make the switch to CES workflow

File: srv/components/ha/cortx-ha/init.sls (hunk 4 above)
Comment (tanujashinde0405) [reply srv/components/ha/cortx-ha/init.sls:22]: Inside the component's overall init.sls , we're calling config.init_mod.sls , which in turn, handles all config phases within. There's no more config.sls file, its deleted and we have only config/ folder inside the components.

File: srv/components/hare/init.sls (hunk 5 above)
Comment (tanujashinde0405) [reply srv/components/hare/init.sls:21]: similar to above implementation

File: srv/components/motr/init.sls (hunk 6 above)
Comment (tanujashinde0405) [reply srv/components/motr/init.sls:22]: similar to above implementation

File: srv/components/s3server/init.sls (hunk 7 above)
Comment (tanujashinde0405) [reply srv/components/s3server/init.sls:22]: similar to above implementation

File: srv/components/sspl/init.sls (hunk 8 above)
Comment (tanujashinde0405) [reply srv/components/sspl/init.sls:22]: similar to above implementation

**Comment by ypise** [comment]
Fix these codacy issues:
https://app.codacy.com/gh/Seagate/cortx-prvsnr/pullRequest?prid=7186149#newIssuesView

**Comment by tanujashinde0405** [comment]
Added new Lines for each end of file  and for 'prepare.sls' creating new PR

**Comment by tanujashinde0405** [comment]
Created another PR including prepare files, closing This PR

**Pull request closed without merging**